### PR TITLE
Send less title changes to the embedder

### LINKED ISSUE #33286: Excessive number of title updates when loading any page
**Describe the bug:**
When loading any page (tested on https://servo.org), the `EmbedderMsg::ChangePageTitle` message is sent many times, incrementally updating the title, even though it never *actually* changes.

Example logs:
```
title: Some("")
title: Some("Servo,")
title: Some("Servo, the embeddable")
title: Some("Servo, the embeddable, independent, memor")
title: Some("Servo, the embeddable, independent, memory-safe, modular")
title: Some("Servo, the embeddable, independent, memory-safe, modular, parallel web r")
title: Some("Servo, the embeddable, independent, memory-safe, modular, parallel web rendering engine")
```

This is because the parser hands out chunks of the elements text incrementally, causing `HtmlTitleElement::children_changed` to be called repeatedly.
While not a big issue, this seems like a silly thing to do.

**To Reproduce:**
apply
```diff
diff --git a/ports/servoshell/desktop/webview.rs b/ports/servoshell/desktop/webview.rs
index a6a33f6a7a..01612f5f9b 100644
--- a/ports/servoshell/desktop/webview.rs
+++ b/ports/servoshell/desktop/webview.rs
@@ -628,6 +628,8 @@ where
                     need_update = true;
                 },
                 EmbedderMsg::ChangePageTitle(title) => {
+                    println!("title: {title:?}");
+
                     // Set the title to the target webview, and update the OS window title
                     // if this is the currently focused one.
                     if let Some(webview_id) = webview_id {

```
and load any page.

**Expected Behaviour**:
One single `EmbedderMsg::ChangePageTitle` once the entire `<title>` element has been parsed.

### PR DESCRIPTION
**Pull request by webbeef**
Instead of sending a title change for each incremental parsing change, we now do it when:
- the element is bound to the tree.
- the parser is done with the element (in pop())
- the title content changes after the parsing, eg. using document.title

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #33286 (GitHub issue number if applicable)

### CODE REVIEW

File: components/script/dom/htmltitleelement.rs [hunk 1]
```diff
@@ -70,9 +74,14 @@ impl VirtualMethods for HTMLTitleElement {
...
-        if node.is_in_doc() {
-            node.owner_doc().title_changed();
+
+        // Notify of title changes only after the initial full parsing
+        // of the element.
+        if self.popped.get() {
+            let node = self.upcast::<Node>();
+            if node.is_in_doc() {
+                node.owner_doc().title_changed();
+            }
```
Comment (Taym95) [inline components/script/dom/htmltitleelement.rs:84]: Maybe create a function for this to avoid duplication?

**Comment by jdm** [comment]
I wonder if we should solve this in the parser instead? I haven't fully wrapped my head around the implications, but I wonder if we could build up a text buffer in the parser and flush it all at once rather than update element contents incrementally.

**Comment by jdm** [comment]
I've filed #33339 about investigating the underlying parser issue. I think this is a reasonable patch over the existing behaviour until we can prioritize that investigation.

**Review by jdm — APPROVED** [review]

**Review by atbrakhi — COMMENTED** [review]
@webbeef just the Signed-off is missing :)

**Pull request merged**